File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int res = 0;
        Set<Integer> set = new HashSet<>();
        for (int num : nums)
            set.add(num);
        for (int num : nums) {
            if (!set.contains(num - 1)) { // start of sequence
                int len = 0;
                while (set.contains(num + len))
                    len++;
                res = Math.max(res, len);
            }
        }
        return res;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        s = set(nums)
        for num in s:
            if num - 1 not in s:
                step = 1
                while num + step in s:
                    step += 1
                res = max(res, step)
        return res